### PR TITLE
Move tests to their own folder ourside `conans`, reorganize some of them

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Omit
Docs: Omit

Took the opportunity to fix a few that were flacky based on your system configuration

### CODE REVIEW

File: test/functional/command/config_install_test.py [hunk 1]
```diff
@@ -481,7 +481,7 @@ def download_verify_true(obj, url, file_path, **kwargs):  # @UnusedVariable
     def test_git_checkout_is_possible(self):
         folder = self._create_profile_folder()
         with self.client.chdir(folder):
-            self.client.run_command('git init .')
+            self.client.run_command('git init . -b master')
```
Comment (AbrilRBS) [inline test/functional/command/config_install_test.py:484]: This was failing if git's `init.defaultBranch` was set to anything other than `master`

File: test/performance/test_large_graph.py [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+from test.assets.genconanfile import GenConanfile
+from test.utils.tools import TestClient
 
 
 def test_large_graph():
-    c = TestClient(cache_folder="T:/mycache")
-    num_test = 40
-    num_pkgs = 40
+    tmp = temp_folder()
+    c = TestClient(cache_folder=tmp)
```
Comment (AbrilRBS) [inline test/performance/test_large_graph.py:13]: This was trying to create a file in my system if I tried to run it - it just created a temp folder now, but is there a way to have it deterministic? The original idea was not to create 40 pkgs each time but have them stored between perfomance runs /cc @memsharded

**Review by AbrilRBS — COMMENTED** [review]
Some minor comments on fixes to the tests

File: test/performance/test_large_graph.py (hunk 2 above)
Comment (memsharded) [reply test/performance/test_large_graph.py:13]: Yes, performance tests are now dirty and broken, more for local development, not intended for CI

The creation of the 40 pkgs takes a long time, so it is commented and hardcoded

**Pull request closed without merging**